exporter/loggingexporter/known_sync_error.go

```diff
@@ -23,14 +23,24 @@ import (
 )
 
 // knownSyncError returns true if the given error is one of the known
-// non-actionable errors returned by Sync on Linux and macOS:
-//
-// Linux:
-// - sync /dev/stdout: invalid argument
-//
-// macOS:
-// - sync /dev/stdout: inappropriate ioctl for device
-//
+// non-actionable errors returned by Sync on Linux and macOS.
 func knownSyncError(err error) bool {
-	return errors.Is(err, syscall.EINVAL) || errors.Is(err, syscall.ENOTSUP) || errors.Is(err, syscall.ENOTTY)
+	var knownSyncErrors = []error{
+		// sync /dev/stdout: invalid argument
+		syscall.EINVAL,
+		// sync /dev/stdout: not supported
+		syscall.ENOTSUP,
+		// sync /dev/stdout: inappropriate ioctl for device
+		syscall.ENOTTY,
+		// sync /dev/stdout: bad file descriptor
+		syscall.EBADF,
+	}
```

Comment:
this can be a "constant" (global var) to avoid initializing every time it is called.